examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,579 @@
+        Number of samples in the testing set.
+
+    Returns
+    -------
+    corrected_std : int
+        Variance-corrected standard deviation of the set of differences.
+    """
+    n = n_train + n_test
+    corrected_var = (
+        np.var(differences, ddof=1) * ((1 / n) + (n_test / n_train))
+    )
+    corrected_std = np.sqrt(corrected_var)
+    return corrected_std
+
+
+def compute_corrected_ttest(differences, df, n_train, n_test):
+    """Computes right-tailed paired t-test with corrected variance.
+
+    Parameters
+    ----------
+    differences : array-like of shape (n_samples, 1)
+        Vector containing the differences in the score metrics of two models.
+    df : int
+        Degrees of freedom.
+    n_train : int
+        Number of samples in the training set.
+    n_test : int
+        Number of samples in the testing set.
+
+    Returns
+    -------
+    t_stat : float
+        Variance-corrected t-statistic.
+    p_val : float
+        Variance-corrected p-value.
+    """
+    mean = np.mean(differences)
+    std = corrected_std(differences, n_train, n_test)
+    t_stat = mean / std
+    p_val = t.sf(np.abs(t_stat), df)  # right-tailed t-test
+    return t_stat, p_val
+
+
+# %%
+model_1_scores = model_scores.iloc[0].values  # scores of the best model
+model_2_scores = model_scores.iloc[1].values  # scores of the second-best model
+
+differences = model_1_scores - model_2_scores
+
+n = differences.shape[0]  # number of test sets
+df = n - 1
+n_train = len(list(cv.split(X, y))[0][0])
+n_test = len(list(cv.split(X, y))[0][1])
+
+t_stat, p_val = compute_corrected_ttest(differences, df, n_train, n_test)
+print(f"Corrected t-value: {t_stat:.3f}\n"
+      f"Corrected p-value: {p_val:.3f}")
+
+# %%
+# We can compare the corrected t- and p-values with the uncorrected ones:
+
+t_stat_uncorrected = (
+    np.mean(differences) / np.sqrt(np.var(differences, ddof=1) / n)
+)
+p_val_uncorrected = t.sf(np.abs(t_stat_uncorrected), df)
+
+print(f"Uncorrected t-value: {t_stat_uncorrected:.3f}\n"
+      f"Uncorrected p-value: {p_val_uncorrected:.3f}")
+
+# %%
+# Using the conventional significance alpha level at `p=0.05`, we observe that
+# the uncorrected t-test concludes that the first model is significantly better
+# than the second.
+#
+# With the corrected approach, in contrast, we fail to detect this difference.
+#
+# In the latter case, however, the frequentist approach does not let us
+# conclude that the first and second model have an equivalent performance. If
+# we wanted to make this assertion we need to use a Bayesian approach.
+
+# %%
+# Comparing two models: Bayesian approach
+# ---------------------------------------
+# We can use Bayesian estimation to calculate the probability that the first
+# model is better than the second. Bayesian estimation will output a
+# distribution that specifies the probability of each parameter value, which in
+# this example is the mean of the differences in the performance of two models.
```

Comment:
Just wondering: is the random variable that we're trying to model really the *mean* (i.e. expectation) of the differences, or is it actually the differences themselves? The description and the rest of the text suggests the former.

Also (almost) unrelated but may I suggest to define mu here already:

> Bayesian estimation will output a distribution that specifies the distribution followed by the mean `` :math:`\mu` `` of the differences in the performance of two models.

(if what we model isn't the average diff but just the diff, then we can just reformulate and maybe switch mu for some other letter)

Reply:
Thanks for noticing this and all of the comments below, the wording is indeed wrong in many places.

Answering the comment here: we are indeed trying to model the __mean__ of the differences.